### PR TITLE
Add forgot password component

### PR DESCRIPTION
**Pull request by MaHaWo**
- add classical 'forgot password' page 
- try to include backend API calls 
- add localization: de, en

### CODE REVIEW

File: frontend/src/lib/components/UserForgotPassword.svelte [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+	import { base } from '$app/paths';
+	import { resetForgotPassword } from '$lib/client/services.gen';
+	import AlertMessage from '$lib/components/AlertMessage.svelte';
+	import DataInput from '$lib/components/DataInput/DataInput.svelte';
+	import { Button, Card, Heading, Input } from 'flowbite-svelte';
+	import { _ } from 'svelte-i18n';
+
+	const maildata = {
+		component: Input,
+		type: 'text',
```
Comment (lkeegan) [inline frontend/src/lib/components/UserForgotPassword.svelte:12]: if you make the type 'email' and make it required then you get validation of the email automatically: see e.g. https://github.com/ssciwr/mondey/blob/main/frontend/src/lib/components/Admin/Login.svelte#L24-L31

File: frontend/src/lib/components/UserForgotPassword.svelte [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+		</div>
+
+		<div class="m-2 flex w-full items-center justify-center p-2">
+			<Button
+				size="md"
+				on:click={(event) => {
+					valid = validateEmail(maildata.value);
+					showAlert = !valid;
+					if (valid) {
+						submitData(maildata.value);
```
Comment (lkeegan) [inline frontend/src/lib/components/UserForgotPassword.svelte:87]: If you wrap the inputs in a form and make the button of type "submit" then the user pressing enter will press the button, e.g.
https://github.com/ssciwr/mondey/blob/d4ea8f85283c81e378988904473b4a3d4cd2e9b7/frontend/src/lib/components/Admin/Login.svelte#L21
https://github.com/ssciwr/mondey/blob/d4ea8f85283c81e378988904473b4a3d4cd2e9b7/frontend/src/lib/components/Admin/Login.svelte#L44

**Review by lkeegan — APPROVED** [review]
lgtm - some minor suggestions

File: frontend/src/locales/de.json [hunk 3]
```diff
@@ -44,5 +44,18 @@
 		"help": "Förderhilfen",
 		"image": "Bild",
 		"images": "Bilder"
+	},
+	"user": {
```
[the remark this answers is not in the document]
Comment (MaHaWo) [reply frontend/src/locales/de.json:48]: thx.

File: frontend/src/lib/components/UserForgotPassword.svelte [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+	async function submitData(mailstring: string): Promise<void> {
+		const data = {
+			body: {
+				email: mailstring
+			},
+			throwOnError: true
+		};
+
+		await resetForgotPassword(data)
```
[the remark this answers is not in the document]
Comment (MaHaWo) [reply frontend/src/lib/components/UserForgotPassword.svelte:41]: ok. I don´t think it's a problem in this specific case imho, but as a general rule it's certainly valid...

**Comment by MaHaWo** [comment]
ok, I made the changes you remarked. you want to have another look @lkeegan  or should I merge this?

**Pull request merged**